### ISSUE TITLE
New Entity/MapObject model.

### ISSUE THREAD

**Issue by ZaneDubya** [ZaneDubya/UltimaXNA#24]
Currently, the Map contains Tiles, which contain MapObjects. Each frame, each Entity adds (or updates?) its MapObject within the associated Tile. Then, on draw, the Tile draws all of its MapObjects.

The dual Entity / MapObject is unnecessary, and an artifact of my implementing Clint's XNA renderer (he used MapObjects, and I pretty much bolted on his code to the client without considering how to re-implement the way it worked to better fit the client's code).

Instead, each Entity should belong to/exist within one MapTile, and on draw, the MapTile should be able to ask the Entity for its 'EntityView' object, which describes how that entity should be drawn.

**In short:**
_Old Model:_
OnUpdate: Each entity creates a MapObject, which is added to Tile.
OnDraw: Tile draws all MapObjects.
_New Model:_
OnUpdate: nothing.
OnDraw: Tile requests a EntityView for each Entity within that Tile, and draws these EntityViews.

**Comment by ZaneDubya** [comment]
This is an example of why the old system sucks:

In BaseEntity.cs:
```c#
void flushDrawObjects()
        {
            if (!Position.IsNullPosition)
            {
                MapTile lastTile = IsometricRenderer.Map.GetMapTile(Position.X, Position.Y, false);
                if (lastTile != null)
                    lastTile.FlushObjectsBySerial(Serial);
            }
        }
```

And in Mobile.cs:
```c#
MapObjectMobile mobtile = new MapObjectMobile(position, DrawFacing, m_animation.ActionIndex, m_animation.AnimationFrame, this);
            tile.AddMapObject(mobtile);

            int[] drawLayers = m_DrawLayerOrder;
			bool hasOuterTorso = m_equipment[(int)EquipLayer.OuterTorso] != null && m_equipment[(int)EquipLayer.OuterTorso].AnimationDisplayID != 0;

            for (int i = 0; i < drawLayers.Length; i++)
            {
				// when wearing something on the outer torso the other torso stuff is not drawn
				if (hasOuterTorso && (drawLayers[i] == (int)EquipLayer.InnerTorso || drawLayers[i] == (int)EquipLayer.MiddleTorso))
				{
					continue;
				}

                if (drawLayers[i] == (int)EquipLayer.Body)
                {
                    mobtile.AddLayer(BodyID, Hue);
                }
                else if (m_equipment[drawLayers[i]] != null && m_equipment[drawLayers[i]].AnimationDisplayID != 0)
                {
                    mobtile.AddLayer(m_equipment[drawLayers[i]].AnimationDisplayID, m_equipment[drawLayers[i]].Hue);
                }
            }
```

So not only are entities responsible for spawning and despawning their MapObjects, but also responsible for Drawing their own overheads. Guh.

**Comment by ZaneDubya** [comment]
Noticed some weird rubber banding with new Position3D code. Probably introduced in f4d6bbf.

**Comment by ZaneDubya** [comment]
Completed in 5753068d3df0d888638f92bff4b4691f491e3c89. Will not merge until deferred rendering is done.
